File: headless_solutions/reordered_power_of_2/solution.js
/**
 * @param {number} n
 * @return {boolean}
 */
var reorderedPowerOf2 = function(n) {
    const sortDigits = (num) => num.toString().split('').sort().join('');
    const target = sortDigits(n);
    
    for (let i = 0; i < 31; i++) {
        if (target === sortDigits(1 << i)) return true;
    }
    return false;
};

File: headless_solutions/reordered_power_of_2/solution.cpp
#include <string>
#include <algorithm>
#include <vector>

class Solution {
public:
    bool reorderedPowerOf2(int n) {
        std::string s = std::to_string(n);
        std::sort(s.begin(), s.end());
        
        for (int i = 0; i < 31; ++i) {
            std::string p = std::to_string(1 << i);
            std::sort(p.begin(), p.end());
            if (s == p) return true;
        }
        return false;
    }
};
